src/main/java/seedu/address/logic/parser/AssignCommandParser.java

```diff
@@ -0,0 +1,27 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.AssignCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+public class AssignCommandParser implements Parser<AssignCommand> {
+    private static final int RESIDENT_INDEX = 0;
+    private static final int EVENT_INDEX = 1;
+
+    @Override
+    public AssignCommand parse(String args) throws ParseException {
+        try {
+            // assign 1 1
+            String[] split = args.stripLeading().split(" ", 2); // { 1, 1 }
+            Index residentIndex = ParserUtil.parseIndex(split[RESIDENT_INDEX]);
+            Index eventIndex = ParserUtil.parseIndex(split[EVENT_INDEX]);
```

Comment:
Hmm... this is okay, but do test with very weird user inputs - like if they insert spaces anywhere.

E.g. `assign 6   1`, `assign  6 1`, etc.

Reply:
Good catch! Thankfully additional spaces seem to have no impact upon testing thanks to `ParserUtil#parseIndex()` as they will trim the input.
But if there any more bugs do report them.

Reply:
Okay, try to keep this in mind while testing 😄